### ISSUE TITLE
[Pay July 23?][$500] mWeb - Chat - Quotes without text are not shown

### ISSUE THREAD

**Issue by lanitochka17** [Expensify/App#42082]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 1.4.73
**Reproducible in staging?:** Y
**Reproducible in production?:** Y
**If this was caught during regression testing, add the test name, ID and link from TestRail:** https://expensify.testrail.io/index.php?/tests/view/4554720
**Issue reported by:** Applause - Internal Team

## Action Performed:
1. Go to https://staging.new.expensify.com/home
2. Tap on a report
3. Paste the text
   
```
> cat

>
>
> has
``` 

4. Send the message

## Expected Result:
Where 
```
> cat

>
>
> has
``` 

should render as 

> cat

>
>
> has

For avoidance of doubt, it's this (screenshot)
<img width="110" alt="image" src="https://github.com/Expensify/App/assets/22508304/1f2cf3f4-aacc-4e06-a6e4-b46d0e18b8dc">

## Actual Result:
Empty quotes at 3rd & 4th line are not shown

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android: Native
- [x] Android: mWeb Chrome
- [ ] iOS: Native
- [ ] iOS: mWeb Safari
- [ ] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos

Add any screenshot/video evidence

</details> 

https://github.com/Expensify/App/assets/78819774/ad43f008-20fa-41ce-8b4c-ac92e65f2388

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~0198eff8474488c59c</li>
        <li>Upwork Job ID: 1793084369808314368</li>
        <li>Last Price Increase: 2024-08-02</li>
    </ul>
</details>

<details><summary>Issue Owner</summary>Current Issue Owner: @situchan</details>

**Comment by lanitochka17** [comment]
@mallenexpensify  FYI I haven't added the External label as I wasn't 100% sure about this issue. Please take a look and add the label if you agree it's a bug and can be handled by external contributors

**Comment by lanitochka17** [comment]
We think that this bug might be related to #vip-vsp

**Comment by mallenexpensify** [comment]
@dannymcclain  👀 plz.  the bug is that this
<img width="223" alt="image" src="https://github.com/Expensify/App/assets/22508304/36d8a401-73b9-4dff-bc46-39c5970663e9">

Renders as
<img width="218" alt="image" src="https://github.com/Expensify/App/assets/22508304/567028d1-2fdd-42eb-8154-9c3b73d42aaa">

So the two lines that only have `>` aren't recognized.  My guess is this is an edge case we can fix on the cheap (I'd start the job at $125). Wanted to check with you before moving forward.  

fwiw... I also tested another version that I thought would be more common and it was rendered as expected

<img width="189" alt="image" src="https://github.com/Expensify/App/assets/22508304/e4f3a795-7793-43be-8810-c557cde745b1">

rendered

<img width="168" alt="image" src="https://github.com/Expensify/App/assets/22508304/74f35ec4-fbf4-44f5-8ea8-39c6d2328247">

**Comment by dannymcclain** [comment]
@mallenexpensify I'm actually a little torn on this. The way ours is working looks pretty identical to the way Github's markdown works... Here's the same situation in GH:

![CleanShot 2024-05-15 at 08 39 05](https://github.com/Expensify/App/assets/7516624/e18849e5-2963-4ed9-97fd-f03d84b1ec4b)

I think we should make the `>` in this situation display the quote block styling:
![image](https://github.com/Expensify/App/assets/7516624/3da40309-a8c5-4430-a85d-5347b04aa6d7)

But I'm unsure if we should make empty `>`s display anything. @Expensify/design what do you think? What would you expect? I'm leaning towards this being fine as is, and just fixing the display of a `>` sandwiched between two other lines.

**Comment by shawnborton** [comment]
Yeah, I think that makes sense to me. So basically as soon as you start typing a `>` on a new line, we convert it to a block quote and add the border on the left?

**Comment by dannymcclain** [comment]
Yeah that's what I'm thinking.

**Comment by dubielzyk-expensify** [comment]
That sounds good to me too

**Comment by mallenexpensify** [comment]
@dannymcclain I wanna update the OP, how would you expect the below to show?

```
> cat

>
>
> has
```

**Comment by dannymcclain** [comment]
Github just gets rid of the two `>` lines above "has", so I'm tempted to follow that pattern.

It basically looks like they ditch any quoted lines unless the lines A) have text or B) are directly sandwiched between two quoted lines with text.

@Expensify/design do you have any other thoughts on how this should work? I don't know that I feel super strongly, just trying to arrive somewhere that makes sense and feels "normal".

**Comment by shawnborton** [comment]
I don't have any strong feelings either. I can see why GH does it the way they do, so I'd be cool with doing something like that.

**Comment by dubielzyk-expensify** [comment]
> Github just gets rid of the two > lines above "has", so I'm tempted to follow that pattern.

Maybe GH has trained me, but that seems to make sense to me too.

**Comment by mallenexpensify** [comment]
Thanks.... everyone on the design team :)

Updated OP and made external 
<img width="847" alt="image" src="https://github.com/Expensify/App/assets/22508304/97f33b1a-ebab-4b2a-b82b-51076d5f33d3">

**Comment by situchan** [comment]
@gijoe0295 this is actual behavior:

![image](https://github.com/Expensify/App/assets/108292595/c4737895-cf8a-4bcb-a020-0824784fb4c6)


Expected behavior:
![image](https://github.com/Expensify/App/assets/108292595/15f629a7-d34d-4827-b98e-1c2b43746417)

**Comment by situchan** [comment]
Waiting for proposals

**Comment by mallenexpensify** [comment]
@situchan , 👀 on the above proposal plz

**Comment by situchan** [comment]
@dominictb thanks for the proposal. Can you please share example of this case - if it's sandwiched between 2 quotes with text?


And I see another change in the beginning, not only + (not mentioned in your proposal).

current regex:
```
/^(?:&gt;)+ +(?! )(?![^<]*(?:<\/pre>|<\/code>))([^\v\n\r]+)/gm;
```

your regex:
```
/^&gt; *(?! )(?![^<]*(?:<\/pre>|<\/code>))([^\v\n\r]*)/gm;
```

**Comment by situchan** [comment]
Awaiting proposals

**Comment by situchan** [comment]
reviewing updated proposal

**Comment by badeggg** [comment]
I also see that it's working as expected now.

**Comment by badeggg** [comment]
I also see that it's working as expected, agree with https://github.com/Expensify/App/issues/42082#issuecomment-2123751943

Plus I noticed some other bugs(maybe):
- If some style need be fixed, it should be the inconsistency of display in composer box and display-after-sent, especially case:
    ```
       > cat
       >
       > has
    ```
- When I copy and paste text that including `\n`, console error appears and send button changed to grey, can not send

video:

https://github.com/Expensify/App/assets/10908996/5fba6872-7c4f-4b17-ab7e-6eeea30be0c4

**Comment by mallenexpensify** [comment]
@situchan , 👀 on the vid above from @badeggg, at the ~50 second mark. Any idea why we're throwing an error there?

**Comment by situchan** [comment]
> If some style need be fixed, it should be the inconsistency of display in composer box and display-after-sent

I think this should be fixed here. So both should look like this:
![image](https://github.com/Expensify/App/assets/108292595/b3a2a5d0-3c6f-40fc-934a-8508d7e38a8a)


> When I copy and paste text that including \n, console error appears and send button changed to grey, can not send

```
> cat
>
> has

```

This bug seems to have another root cause and can be out of scope

**Comment by mallenexpensify** [comment]
@dominictb can you report the other bug in #expensify-open-source and tag me? Start with 
```
Coming from [here](https://github.com/Expensify/App/issues/42082) cc @mattallen
BUG:   
Action Performed: 
Expected Result:
Actual Result:
Platforms:
Version Number:
Screenshots or Videos:  
```
I'll give you first dibs to propose a solution if we decide we want to fix it.

**Comment by situchan** [comment]
@dominictb please make sure that your solution passes automated test. `npm run test` in expensify-common. There are 3 failures.
![image](https://github.com/Expensify/App/assets/108292595/6007ab66-dde2-4d21-9b25-e8817c141aee)

**Comment by situchan** [comment]
@dominictb mind sharing your test branch?

**Comment by mallenexpensify** [comment]
> @dominictb can you report the other bug in #expensify-open-source and tag me? Start with

d'oh, def got that wrong, thx for letting me know.  

@badeggg can you report [the other bug](https://github.com/Expensify/App/issues/42082#issuecomment-2142459123) in #expensify-open-source and tag me? Start with 
```
Coming from [here](https://github.com/Expensify/App/issues/42082) cc @mattallen
BUG:   
Action Performed: 
Expected Result:
Actual Result:
Platforms:
Version Number:
Screenshots or Videos:  
```
I'll give you first dibs to propose a solution if we decide we want to fix it.

**Comment by badeggg** [comment]
@mallenexpensify [bug reported](https://expensify.slack.com/archives/C01GTK53T8Q/p1717559130850559)

**Comment by situchan** [comment]
still looking for acceptable proposal

**Comment by mallenexpensify** [comment]
Bumped this to $500 to try to get it over the finish line, since it's a month old.

**Comment by situchan** [comment]
Waiting for proposals

**Comment by situchan** [comment]
>as soon as you start typing a > on a new line, we convert it to a block quote and add the border on the left?

@Expensify/design when there's no space between > and text, which one should we go with? 

```
>test
```

Current app: (not considered as quote)

```
>test
```

After fix: (same behavior as github)

>test

**Comment by dannymcclain** [comment]
I prefer the after fix behavior personally (same as Github). Happy for others to chime in, but I made this mistake earlier thinking it would work and had to edit my message for it get styled properly.

So I think
`>this`
should render like
>this

**Comment by situchan** [comment]
@dominictb please update your proposal based on latest discussion.
Your regex doesn't work for me.

**Comment by badeggg** [comment]
Hi, what this comment should be rendered in composer box:
```
> cat

>
>
> has
```
I know it should be rendered like this in chat history:
<img width="200" alt="image" src="https://github.com/Expensify/App/assets/10908996/296b55f8-46d5-4d34-861f-a9b08ff92ea6">

**Comment by sobitneupane** [comment]
@situchan Will [this issue](https://github.com/Expensify/App/issues/43323) also be covered here?

**Comment by badeggg** [comment]
@sobitneupane Seems https://github.com/Expensify/App/issues/42082 and https://github.com/Expensify/App/issues/43323 are different

**Comment by badeggg** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
Inconsistency of display for same comment in composer box and chat history. Specifically:
```
> cat
>
> has
```
And we should recognize `>a` as a quote block.

### What is the root cause of that problem?
Regex [`const regex = /^(?:&gt;)+ +(?! )(?![^<]*(?:<\/pre>|<\/code>))([^\v\n\r]+)/gm;`](https://github.com/Expensify/expensify-common/blob/c6d1c00652f016532e33ad47c98244538247890b/lib/ExpensiMark.js#L262) and [`const rawInputRegex = /^(?:&gt;)+ +(?! )(?![^<]*(?:<\/pre>|<\/code>))([^\v\n\r]*)/gm;`](https://github.com/Expensify/expensify-common/blob/c6d1c00652f016532e33ad47c98244538247890b/lib/ExpensiMark.js#L265) are not recognizing `>no-space` as a quote block.

### What changes do you think we should make in order to solve the problem?

We should change [regex](https://github.com/Expensify/expensify-common/blob/c6d1c00652f016532e33ad47c98244538247890b/lib/ExpensiMark.js#L262C21-L262C98) from:
```js
const regex = /^(?:&gt;)+ +(?! )(?![^<]*(?:<\/pre>|<\/code>))([^\v\n\r]+)/gm;
```
to
```js
const regex = /^(?:&gt;)+ *(?! )(?![^<]*(?:<\/pre>|<\/code>))([^\v\n\r]+)/gm;
```
, from matching '1 or more' spaces to matching '0 or more' spaces.

Similar change should be applied to [this regex](https://github.com/Expensify/expensify-common/blob/c6d1c00652f016532e33ad47c98244538247890b/lib/ExpensiMark.js#L265) too.

**Notice:**
1. This change is same as [recommended change in this comment I made](https://github.com/Expensify/App/issues/43323#issuecomment-2172147184)
2. 
    ```
    > cat

    >
    >
    > has
    ```
    will be rendered in composer box like this:
<img width="403" alt="image" src="https://github.com/Expensify/App/assets/10908996/49ef529d-77e5-400c-b506-e6d3da8fc397">


### What alternative solutions did you explore? (Optional)
N/A

**Comment by badeggg** [comment]
@situchan About [this comment](https://github.com/Expensify/App/issues/42082#issuecomment-2125609465), can you share your app version that produce display result in your comment? I can not see that result.

To be clear, from my view, for:
```
> cat

>
>
> has
```
Currently, in composer box, it's rendered as: 
<img width="558" alt="image" src="https://github.com/Expensify/App/assets/10908996/bcdf4882-ee3a-4650-b5fd-bcac0fbb1125">
And after sent, rendered as(which is what we want):
<img width="425" alt="image" src="https://github.com/Expensify/App/assets/10908996/4f0ee5ca-7437-4cbb-b902-fdd393871ca9">

**Comment by situchan** [comment]
> @situchan Will [this issue](https://github.com/Expensify/App/issues/43323) also be covered here?

@sobitneupane yes

**Comment by situchan** [comment]
> @situchan About [this comment](https://github.com/Expensify/App/issues/42082#issuecomment-2125609465), can you share your app version that produce display result in your comment? I can not see that result.
> 
> To be clear, from my view, for:
> 
> ```
> > cat
> 
> >
> >
> > has
> ```
> 
> Currently, in composer box, it's rendered as: <img alt="image" width="558" src="https://private-user-images.githubusercontent.com/10908996/340271088-bcdf4882-ee3a-4650-b5fd-bcac0fbb1125.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTg2NTI0MzcsIm5iZiI6MTcxODY1MjEzNywicGF0aCI6Ii8xMDkwODk5Ni8zNDAyNzEwODgtYmNkZjQ4ODItZWUzYS00NjUwLWI1ZmQtYmNhYzBmYmIxMTI1LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA2MTclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNjE3VDE5MjIxN1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWNiZTczMmU3MmI0NDFlNTM1YWUwYzAyOWUzYTJkZjNmOTEzMDQzZWY4MTJiMDc5NTg1Y2YwY2U0NWI3MjE5NmYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.7BXX1Q_IXjBOv__HjawSfFClFFKV7E7Vocv7rqm8xN0"> And after sent, rendered as(which is what we want): <img alt="image" width="425" src="https://private-user-images.githubusercontent.com/10908996/340271245-4f0ee5ca-7437-4cbb-b902-fdd393871ca9.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTg2NTI0MzcsIm5iZiI6MTcxODY1MjEzNywicGF0aCI6Ii8xMDkwODk5Ni8zNDAyNzEyNDUtNGYwZWU1Y2EtNzQzNy00Y2JiLWI5MDItZmRkMzkzODcxY2E5LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA2MTclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNjE3VDE5MjIxN1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWQ0N2MzNjZkM2JlMzVkMDRhN2NkYTUxZmIzZmE1ZTJlYWFiMjYxNDE3OGRmZTY0NzM4NzE4YmM3MDI3NmUxNDYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.OEhyJFx2ZKcurQa-i6jsmJU8eg7jwEYkMZzV9Ac4HbA">

@badeggg I tested on latest staging v1.4.84-2
![image](https://github.com/Expensify/App/assets/108292595/502b2023-3ec5-49ff-bbf9-42101989265c)

**Comment by badeggg** [comment]
I am withdrawing my proposal for this issue, since I can not reproduce this :
<img width="545" alt="image" src="https://github.com/Expensify/App/assets/10908996/8a296caf-e7a6-4d06-962f-443941736546">

**Comment by BrtqKr** [comment]
Might be related to what I've sent [here](https://github.com/Expensify/expensify-common/pull/717). I'll verify it.

I'm a developer from SWM, I'd like to take over this issue.

**Comment by BrtqKr** [comment]
Ok, to sum up:
Some time ago[ I enabled display of the empty blockquotes](https://github.com/Expensify/App/issues/41107), which covered the case of empty lines not appearing in the case presented below

<img width="500" alt="image" src="https://github.com/Expensify/App/assets/45288762/1b857515-1200-4c95-98d2-98167c9ca747">

But one issue was left for a separate PR, which was basically about the empty lines being removed from the edges of the blockquote container, which in result was completely messing up the display

<img width="500" alt="image" src="https://github.com/Expensify/App/assets/45288762/df2a34b9-7b5a-4820-bb24-6699580bde74">


On the most recent `main`, because of this trim the result would look like this:

<img width="407" alt="image" src="https://github.com/Expensify/App/assets/45288762/b9b6feb8-a0e7-4589-ad83-22922a5df7f6">

and this is with the changes from the PR I've mentioned above

<img width="310" alt="image" src="https://github.com/Expensify/App/assets/45288762/8ecc7c07-50f5-4178-ba00-d87dbcd61408">


So I believe this should no longer be relevant. I'll finish the PR tomorrow

**Comment by situchan** [comment]
@BrtqKr please tag me when app PR which bumps expensify-common is ready for review

**Comment by BrtqKr** [comment]
@situchan I've bumped this inside of live-markdown, here's the PR for the expensify/app:  https://github.com/Expensify/App/pull/44262

**Comment by mallenexpensify** [comment]
Looks like the PR hit production 3 days ago, updated title and labels, let me know if they don't look correct @situchan

**Comment by mallenexpensify** [comment]
@situchan can you please accept the job and reply here once you have?
[upwork.com/jobs/~0182a80fe9e36869f3](https://www.upwork.com/jobs/~0182a80fe9e36869f3)

**Comment by situchan** [comment]
Not overdue

**Comment by situchan** [comment]
@mallenexpensify accepted thanks

**Comment by mallenexpensify** [comment]
Contributor+: @sobitneupane paid $500 via Upwork. 

@sobitneupane plz complete the BZ checklist above. 

**BugZero Checklist:** The PR fixing this issue has been merged! The following checklist ([instructions](https://stackoverflow.com/c/expensify/questions/14574)) will need to be completed before the issue can be closed:

- [ ] [@] The PR that introduced the bug has been identified. Link to the PR:
- [ ] [@] The offending PR has been commented on, pointing out the bug it caused and why, so the author and reviewers can learn from the mistake. Link to comment:
- [ ] [@] A discussion in #expensify-bugs has been started about whether any other steps should be taken (e.g. updating the PR review checklist) in order to catch this type of bug sooner. Link to discussion:
- [ ] [@] Determine if we should create a regression test for this bug.
- [ ] [@] If we decide to create a regression test for the bug, please propose the [regression test](https://github.com/Expensify/App/blob/main/contributingGuides/REGRESSION_TEST_BEST_PRACTICES.md) steps to ensure the same bug will not reach production again.
- [ ] [@BZ] Link the GH issue for creating/updating the regression test once above steps have been agreed upon: 
- [ ]

**Comment by mvtglobally** [comment]
Issue not reproducible during KI retests. (Second week)

**Comment by situchan** [comment]
There's no offending PR. This is an edge case we didn't consider.
Looks like we already have regression test for this - https://expensify.testrail.io/index.php?/tests/view/4554720

**Comment by mallenexpensify** [comment]
Thanks @situchan
